### ISSUE TITLE
Values are parsed as dates even though they contain extra characters

### ISSUE THREAD

**Issue by tsvetomir** [telerik/kendo-ui-core#1909]
Copying the value in B1 to another cell converts it to a date. This is not happening with the value in A1.
The parsing seems to be a bit too eager in this case.

[Dojo Snippet](http://dojo.telerik.com/isERE/2)

``` html
<div id="spreadsheet"></div>
<script>
    $("#spreadsheet").kendoSpreadsheet({
        sheets: [{
            name: "Sheet1",
            columns:[
              {width:200},
              {width:200}
            ],
            rows: [{
                cells: [{
                    value: "1197-1-000-10.step"
                },
                {
                    value:"1197-1-095-10.step"
                }]
            }]
        }]
    });
</script>
```

**Comment by mishoo** [comment]
The parser relies on `kendo.parseDate` to detect dates, and `kendo.parseDate("1197-1-095-10.step")` does return a Date object.  I suggest the fix should go there instead, as the string obviously doesn't look like a date.

**Comment by MaceWindu** [comment]
as a workaround you can patch parseExact function in kendo.core.js:
before this line: 
```JS
hasTime = hours !== null || minutes !== null || seconds || null;
```
add
```JS
if (valueIdx < value.length) {
    return null;
}
```
or if you want to ignore trailing whitespaces:
```JS
if (valueIdx < value.length && !/^\s*$/.test(value.substring(valueIdx))) {
    return null;
}
```

**Comment by mishoo** [comment]
Fixed sometime ago (see telerik/kendo-ui-core#3476).
